projects/openapi-utilities/src/openapi/openapi2/openapi-2-traverser.ts

```diff
@@ -0,0 +1,891 @@
+import {
+  OpenApiKind,
+  OpenApiRequestParameterFact,
+  OpenApiFieldFact,
+  OpenApiOperationFact,
+  Traverse,
+  OperationLocation,
+  ResponseHeaderLocation,
+  RequestLocation,
+  ResponseLocation,
+  PathParameterLocation,
+  HeaderParameterLocation,
+  BodyLocation,
+  BodyExampleLocation,
+  QueryParameterLocation,
+  FieldLocation,
+  ComponentSchemaLocation,
+  IFact,
+  OpenApiSpecificationFact,
+  FactVariant,
+  OpenApi3SchemaFact,
+  OpenApiRequestFact,
+} from '../sdk/types';
+import { IPathComponent } from '../sdk/types';
+import invariant from 'ts-invariant';
+import { jsonPointerHelpers as jsonPointer } from '@useoptic/json-pointer-helpers';
+import { OpenAPIV2, OpenAPIV3 } from 'openapi-types';
+import {
+  normalizeOpenApiPath,
+  isObject,
+  getReadableLocation,
+  isNotReferenceObject,
+} from '../../utilities/traverse-helpers';
+
+export class OpenAPI2Traverser implements Traverse<OpenAPIV2.Document> {
+  format = 'openapi2';
+
+  input: OpenAPIV2.Document | undefined = undefined;
+
+  traverse(input: OpenAPIV2.Document): void {
+    this.input = input;
+  }
+
+  // see https://github.com/OAI/OpenAPI-Specification/issues/1259
+  getConsumesContentTypes(operation: OpenAPIV2.OperationObject): string[] {
```

Comment:
see https://github.com/OAI/OpenAPI-Specification/issues/1259

consumes = content type for request bodies. It is set in the operation || globally || []